torch/_dynamo/variables/user_defined.py

```diff
@@ -291,9 +291,22 @@ def call_function(
             and len(kwargs) == 0  # TODO(ybliang): support kwargs
         ):
             unwrapped_args = [x.as_python_constant() for x in args]
-            return GenericContextWrappingVariable(
-                unwrapped_args,
-                cm_obj=self.value(*unwrapped_args),
+            if self.value.__init__ == object.__init__:
+                return GenericContextWrappingVariable(
+                    unwrapped_args,
+                    cm_obj=self.value(*unwrapped_args),
+                )
+
+            try:
+                fn = variables.UserFunctionVariable(
+                    self.value.__init__, source=self.source
+                )
```

Comment:
It seems weird to me to construct a UserFunctionVariable directly, shouldn't we call builder on the type of value, and then call_function from there to handle the cases in `UserDefinedClass`?

At the very least, we should guard here unless the guards were already taken care of when this var was built.